modules/aws/lb/main.tf

```diff
@@ -0,0 +1,272 @@
+  is_network     = var.load_balancer_type == "network"
+}
+
+# Load Balancer
+resource "aws_lb" "load_balancer" {
+  # Common settings
+  name                       = var.name
+  internal                   = var.internal
+  load_balancer_type         = var.load_balancer_type
+  subnets                    = var.subnets
+  enable_deletion_protection = var.enable_deletion_protection
+  customer_owned_ipv4_pool   = var.customer_owned_ipv4_pool
+  ip_address_type            = var.ip_address_type
+
+  # Application Load Balancer specific settings
+  security_groups            = local.is_application ? var.security_groups : null
+  desync_mitigation_mode     = local.is_application ? var.desync_mitigation_mode : null
+  idle_timeout               = local.is_application ? var.idle_timeout : null
+  drop_invalid_header_fields = local.is_application ? var.drop_invalid_header_fields : null
+  enable_http2               = local.is_application ? var.enable_http2 : null
+  enable_waf_fail_open       = local.is_application ? var.enable_waf_fail_open : null
+
+  # Network Load Balancer specific settings
+  enable_cross_zone_load_balancing = local.is_network ? var.enable_cross_zone_load_balancing : null
+
+  dynamic "access_logs" {
+    for_each = var.access_logs != null ? [var.access_logs] : []
+    content {
+      bucket  = access_logs.value.bucket
+      prefix  = access_logs.value.prefix
+      enabled = access_logs.value.enabled
+    }
+  }
+
+  dynamic "subnet_mapping" {
+    for_each = var.subnet_mappings
+    content {
+      subnet_id            = subnet_mapping.value.subnet_id
+      allocation_id        = subnet_mapping.value.allocation_id
+      private_ipv4_address = subnet_mapping.value.private_ipv4_address
+      ipv6_address         = subnet_mapping.value.ipv6_address
+    }
+  }
+
+  tags = merge(
+    var.tags,
+    {
+      Name = var.name
+    },
+  )
+}
+
+resource "aws_lb_target_group" "target_group" {
+  for_each = var.target_groups
+
+  # Common settings
+  name                 = var.target_group_name
+  name_prefix          = var.target_group_name_prefix
+  port                 = var.target_group_port
+  protocol             = var.target_group_protocol
+  vpc_id               = var.target_group_vpc_id
+  target_type          = var.target_group_target_type
+  deregistration_delay = var.target_group_deregistration_delay
+
+  # Application Load Balancer specific settings
+  slow_start                    = local.is_application ? var.target_group_slow_start : null
+  load_balancing_algorithm_type = local.is_application ? var.target_group_load_balancing_algorithm_type : null
+
+  # Network Load Balancer specific settings
+  proxy_protocol_v2  = local.is_network ? var.target_group_proxy_protocol_v2 : null
+  preserve_client_ip = local.is_network ? var.target_group_preserve_client_ip : null
+
+  dynamic "health_check" {
+    for_each = each.value.health_check != null ? [each.value.health_check] : []
```

Comment:
Change this to a map for consistency with the type:

```
for_each = each.value.health_check != null ? each.value.health_check : {}
```

Reply:
id now map of object.